core/App/ehsm_napi.cpp

```diff
@@ -1125,4 +1128,64 @@ char* NAPI_ExportDataKey(const char* cmk_base64,
     return retJsonObj.toChar();
 }
 
-}  // extern "C"
+
+    char *NAPI_RA_HANDSHAKE_MSG0(const char *p_msg0)
+    {
+        print_d("***NAPI_RA_HANDSHAKE_MSG0 start.");
+
+        print_d("msg0: \n %s", p_msg0);
+
+        RetJsonObj retJsonObj;
+        retJsonObj.addData("challenge_base64", "challenge_base64");
+        retJsonObj.addData("ga_base64", "ga_base64");
+        print_d("msg1: \n%s",retJsonObj.toChar());
+
+        print_d("***NAPI_RA_HANDSHAKE_MSG0 end.");
+        return retJsonObj.toChar();
+    }
+
+    char *NAPI_RA_HANDSHAKE_MSG2(const char *p_msg2)
+    {
+        print_d("***NAPI_RA_HANDSHAKE_MSG2 start.");
+
+        print_d("msg2: \n %s", p_msg2);
+
+        RetJsonObj retJsonObj;
+        retJsonObj.addData("msg3_base64", "msg3_base64");
+        print_d("msg3: \n%s",retJsonObj.toChar());
+
+        print_d("***NAPI_RA_HANDSHAKE_MSG2 end.");
+        return retJsonObj.toChar();
+    }
+
+
+
+    /*
+     *   Create an UUID
+     */
+    static std::string create_uuid()
+    {
+        uuid_t uu;
+        uuid_generate(uu);
+        char uuid_str[37];
+        uuid_unparse_lower(uu, uuid_str);
+        return uuid_str;
+    }
+
+    char *NAPI_RA_GET_API_KEY(const char *p_msg4)
+    {
+        print_d("***NAPI_RA_GET_API_KEY start.");
+
+        print_d("msg4: \n %s", p_msg4);
+
+        RetJsonObj retJsonObj;
+        retJsonObj.addData("nonce", create_uuid());
+        retJsonObj.addData("appid", create_uuid());
+        retJsonObj.addData("appkey", create_uuid());
```

Comment:
for the app key, you can call the createkey API to generate a aes_gcm_128 key which used as the app access key

Reply:
It will be fix